src/lib/healthCheck.js

```diff
@@ -0,0 +1,89 @@
+/*
+    Health check support for PWA etc.
+*/
+class HealthChecker {
+  /*
+  p - object with parameters:
+      requester - object of library popsicle or vue-resource,
+          should provide method get() -> Promise
+      urls - [{ip, protocol, port}]
+      onStatusChange - callback to be called with args:
+          urls - [{ip, protocol, port, online}]
+      probeDelay - interval of probing, seconds, default 60
+  */
+  constructor (p) {
+    this.requester = p.requester
+    this.urls = p.urls || []
+    this.onStatusChange = p.onStatusChange
+    this.probeDelay = p.probeDelay || 60
+    this.lastStatusList = this.urls.map(o => true)
+    this.intId = null
+  }
+
+  // Start probing
+  start () {
+    this.stop()
+    this.probe()
+    this.intId = setInterval(
+      this.probe.bind(this),
+      this.probeDelay * 1000)
+  }
+
+  // Stop probing
+  stop () {
+    if (this.intId) {
+      clearInterval(this.intId)
+      this.intId = null
+    }
+  }
+
+  // Probe given urls
+  probe () {
+    // Wait for all probes
+    Promise.all(this.urls.map(o => new Promise((resolve, reject) => {
```

Comment:
Maybe better replace 
````o => new Promise((resolve, reject) => {
      this.requester.get(makeUrl(o))
        .then(() => resolve(true))
        .catch(() => resolve(false))
   })
````
with 
```` o => this.requester.get(makeUrl(o)).then(() => true, () => false) ````

Reply:
Well, without call to resolve() it will not work as designed.
I have just tested this thing.